### ISSUE TITLE
Apply Slope Difficulty filter does not consider the selection

### ISSUE THREAD

**Issue by davidobber** [brenzy/gpx-smoother-vue#25]
I have a gpx with a climb and a descent, I would like to apply  the Slope difficulty filter only to a climb.

**Comment by brenzy** [comment]
@davidobber If you drag the arrows in the second graph to highlight your climb, it should work.

Please note that the slope difficulty clamps the percentage increase or decrease in slope.  So for instance, if there is a jump from 1% to 12% and the the slope difficulty is 3, it will only jump from 1% to 4%.  This was added to prevent huge jumps that the trainer can't handle.  If you want to keep the slope at 3% or below, use "Apply Slope Range".  This was added because I hate climbing! 

Please let me know if that works for you, and if so where the documentation should be updated.  If not please send me your file, so I can test.

**Comment by davidobber** [comment]
Hi and thanks for your quick reply.

> This was added because I hate climbing!   😁

It seems to me that Slope difficulty is something else. It's descirbed as
"Percentage increase or decrease in slope" and it does what it says. I've
selected the range with the arrows. However, I tried another time and now
it works. It seems to me that maybe there was some strange state. But now
I've just selected a range and applied the "Flatten Values" filter. But all
the gpx was affected by the change.

Does it happen to you?

David

Il giorno ven 5 feb 2021 alle ore 12:03 brenzy <notifications@github.com>
ha scritto:
